### PR TITLE
Blog app: add devise

### PR DESCRIPTION
**Pull request by KingsleyIbe**
### In this milestone, @KingsleyIbe and @Tchilo implemented the following:

- Removed the current_user method in ApplicationController.
- [Install and setup devise](https://github.com/heartcombo/devise#getting-started).
- App can register a new user.
- User can log in with a combination of email and password.
- Hashed passwords should be stored in the database.
- App asks the user for confirmation of email.
- User can reset the password.

### CODE REVIEW

File: app/views/devise/sessions/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<h2>Log in</h2>
+
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+
+  <div class="field">
+    <%= f.label :Username %><br />
+    <%= f.text_field :name, autofocus: true, autocomplete: "name" %>
+  </div>
```
Comment (TedLivist) [inline app/views/devise/sessions/new.html.erb:8]: - [ ] After signing up I am unable to log in. I keep getting an invalid email and password error message despite your form asking for a Username and password. Kindly fix this.

  I would recommend you refactor this input to the email input you had earlier.
  ```
  <div class="field">
    <%= f.label :email %><br />
    <%= f.text_field :email, autofocus: true, autocomplete: "email" %>
  </div>
  ```

  king-username

File: app/views/devise/sessions/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <%= f.label :remember_me %>
+    </div>
+  <% end %>
+
+  <div class="actions">
+    <%= f.submit "Log in" %>
+  </div>
+<% end %>
+
+<%= render "devise/shared/links" %>
```
Comment (TedLivist) [inline app/views/devise/sessions/new.html.erb:29]: - [ ] You are yet to implement the wireframe layout as requested earlier. Kindly do that.

  Your Implementation | Wireframe
  :---:|:---:
  king-1 | king-2

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @KingsleyIbe,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/devise/sessions/new.html.erb (hunk 2 above)
Comment (KingsleyIbe) [reply app/views/devise/sessions/new.html.erb]: Hello @TedLivist 

Thank you for reviewing my work. Be sure changes will be implemented.

File: app/views/layouts/_header.html.erb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<header>
+  <div>
+    <%= link_to "Blog App", root_path, id: "logo" %>
+ <nav>
+      <ul>
+        <% if user_signed_in? %>
+          <li><%= link_to "Home", root_path %></li>
+          <li><%= link_to "New post", new_post_path %></li>
```
Comment (julie-ify) [inline app/views/layouts/_header.html.erb:8]: - [ ] **[OPTIONAL]** This link `New post` crashes when I click on it to create a new post. You can run `rails routes` to see all your action routes and then pick a route for creating a new post. Also, make sure that the form for creating a new post has the correct URL.

chrome_E3Z64qExwn

Comment (KingsleyIbe) [reply app/views/layouts/_header.html.erb:8]: Hey @julie-ify 

Thank you for reviewing and approving my work.

**Pull request merged**